src/library_sockfs.js

```diff
@@ -280,6 +280,13 @@ mergeInto(LibraryManager.library, {
 
         function handleMessage(data) {
           assert(typeof data !== 'string' && data.byteLength !== undefined);  // must receive an ArrayBuffer
+
+          // An empty ArrayBuffer will emit a pseudo disconnect event
```

Comment:
To be honest, I don't know. My original though was that we should deal with this as early as possible because it doesn't make sense that we pass the received data to the `sock.recv_queue` and process it later. Also it seems to me that these kind of packets shouldn't be even processed by further routines like recvmsg/... .
So there is not much rationale behind it. Do you think it's wrong?